content/guidelines/choosing-selectors.en.md

```diff
@@ -0,0 +1,167 @@
+
+It is possible to reduce the frequency of human intervention on the selectors by choosing some that are least likely to become obsolete even when changes occur on the document structure. Decreasing the cost of this maintenance increases the likelihood that terms are continuously tracked.
+
+## Guidelines
+
+While there is no single right way to choose durable selectors, as it remains intrinsically subjective and dependent on the document itself, following the guidelines below is likely to yield durable selectors.
+
+### Simple
+
+Prefer [simple selectors](https://www.w3.org/TR/selectors/#simple) over compound ones, and compound ones over [combinator](https://www.w3.org/TR/selectors/#combinators) ones, as they are more readable and make less assumptions on future changes.
+
+In particular, avoid deep nesting of elements and pseudo-classes. Such selectors rely not only on the targeted content structure but also on the content around it. The likelihood that at least one block in the tree changes on a page update increases, making the selector brittle.
+
+#### Examples
+
+- `#content` or `[role="main"]` are better than `#content .inner` or `[role="main"] > .content-block`.
+- `main > div > #article > .tos` is a bad selector because of deep nesting.
+- `div:nth-child(2)` is a bad selector because of the usage of a pseudo-class.
+
+### Higher specificity
+
+Prefer selectors with higher [specificity](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity), as they are less likely to select other content when the rest of the document changes.
+
+#### Examples
+
+- `#article .section` (with 1-1-0 specificity) is better than `body .section .tos` (with 0-2-1 specificity).
+
+### Named
+
+Prefer selectors whose names are representative of the parts they select, as they are evidence of an intention by the service provider, and thus more likely to stay valid through design changes.
+
+Conversely, avoid class names being or containing series of alphanumeric characters. Those are most likely to be generated and to change on the next page update.
+
+#### Examples
+
+- `.tos` or `#legal-notice` are better than `main` or `.content`.
+- `.dez68h` or `.tos-cpoxw27` are bad selectors because they are likely automatically generated.
+
+### Accuracy over durability, durability over exhaustiveness
+
+It should be clear that the most important point of a declaration is to track entire terms. Legitimate considerations on selector durability can not be used as arguments justifying not tracking content that impacts users rights and duties.
+
+On the other hand, content that is usually tracked but is not directly impactful on rights and duties, such as last updated date or document title, may be occasionally ignored if it makes selectors excessively brittle.
```

Comment:
I not really agree with that. I'm in favor of having an array of selectors for these cases instead of not tracking this content as I consider it important. So for example, as I said in [this comment](https://github.com/OpenTermsArchive/GenAI-declarations/pull/2/commits/6ad91a4eaee971ba545fd31cb3bc6a83a0b5e884#r1343745906), I prefer :
```
"select": [
  "#landings-terms-of-use"
],
"remove": [
  "header",
  "[aria-hidden=true]",
  "footer"
],
```
instead of:
```
"select": [
  "#content"
],
```

Reply:
I agree with that stated preference.

However, I prefer:

```json
"select": [
  "#inner-container"
]
```

instead of:

```json
"select": [
  "body"
],
"remove": [
  ".af23jf45",
  ".xlms051"
]
```

…even if the former meant not tracking the document title, for example.

If you agree with this, could you suggest an improvement in the phrasing so that this is clearer? 🙂

Reply:
I added an example based on this conversation to clarify our intention here 🙂